dronekit/__init__.py

```diff
@@ -1161,7 +1161,8 @@ def listener(self, name, m):
         def listener(self, name, m):
             self._armed = (m.base_mode & mavutil.mavlink.MAV_MODE_FLAG_SAFETY_ARMED) != 0
             self.notify_attribute_listeners('armed', self.armed, cache=True)
-            self._flightmode = {v: k for k, v in self._master.mode_mapping().items()}[m.custom_mode]
+            if self._master.mode_mapping() != None:
```

Comment:
(just an FYI for anyone else looking at this) `None` here makes sense since upstream pymavlink `mode_mapping` is expected to return `None` https://github.com/mavlink/mavlink/blob/master/pymavlink/mavutil.py#L481-L502